helpers.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+
+const { AUTH_KEYS_KEYS } = require('@mtproto/core');
+
+function extractMessagesFromUpdates({updates, channelId}) {
+  const messages = [];
+  updates
+    // TODO handle update and delete messages
+    .filter(update => update._ === 'updateNewChannelMessage' && update.message.message)
+    .forEach(({message}) => {
+      if (channelId && +message.to_id.channel_id === +channelId) messages.push(message);
+      else if (!channelId) messages.push(message);
+    });
+  return messages;
+}
```

Comment:
it seems you are doing two kinds of filters. Is this the same?:
```js
function extractMessagesFromUpdates({updates, channelId}) {
  if (channelId) channelId = Number(channelId)
  // TODO handle update and delete messages
  return updates
    .filter(update => 
      update._ === 'updateNewChannelMessage' && 
      update.message.message &&
      (
        !channelId || 
        Number(update.message.to_id.channel_id) === channelId
      )
    )
    .map(update => update.message)
}
```

Reply:
Yes, thats the same and much cleaner